File: src/cycod/CommandLineCommands/PromptCommands/PromptBaseCommand.cs
using System;

/// <summary>
/// Base class for all prompt-related commands.
/// </summary>
abstract class PromptBaseCommand : Command
{
    /// <summary>
    /// The scope to use for the prompt operation.
    /// </summary>
    public ConfigFileScope? Scope { get; set; }

    /// <summary>
    /// Constructor initializes the base command.
    /// </summary>
    public PromptBaseCommand()
    {
    }

    /// <summary>
    /// Indicates if the command is empty.
    /// </summary>
    /// <returns>False, as prompt commands are never empty.</returns>
    public override bool IsEmpty()
    {
        return false;
    }
}

File: ts/src/CommandLineCommands/PromptCommands/PromptBaseCommand.ts
import { ConfigFileScope } from '../../Configuration/ConfigFileScope';

/**
 * Base class for all prompt-related commands.
 */
export abstract class PromptBaseCommand {
  public scope?: ConfigFileScope;

  constructor() {
    // Base constructor
  }

  /**
   * Gets the command name.
   */
  abstract getCommandName(): string;

  /**
   * Executes the command.
   */
  abstract executeAsync(interactive: boolean): Promise<number>;

  /**
   * Indicates if the command is empty.
   */
  isEmpty(): boolean {
    return false;
  }
}